shipment/api/let_me_ship.py

```
@@ -305,23 +307,58 @@ def create_letmeship_shipment(
                                       auth=(service_provider.api_key,
                                             service_provider.api_password), headers=headers,
                                       data=json.dumps(payload))
-        response_data = json.loads(response_data.text)
+        
+        # Check if response is valid before parsing JSON
+        if not response_data or not response_data.text:
+            frappe.log_error(f"Empty response from LetMeShip API")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)